drivers/clock_control/clock_control_r8a779f0_cpg_mssr.c

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright (c) 2023 EPAM Systems
+ * Copyright (c) 2023 IoT.bzh
+ *
+ * r8a779f0 Clock Pulse Generator / Module Standby and Software Reset
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT renesas_r8a779f0_cpg_mssr
+
+#include <errno.h>
+#include <zephyr/arch/cpu.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/clock_control/renesas_cpg_mssr.h>
+#include <zephyr/dt-bindings/clock/renesas_cpg_mssr.h>
+#include <zephyr/dt-bindings/clock/r8a779f0_cpg_mssr.h>
+#include <zephyr/irq.h>
+#include "clock_control_renesas_cpg_mssr.h"
+
+#define LOG_LEVEL CONFIG_CLOCK_CONTROL_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_DECLARE(clock_control_rcar);
+
+struct r8a779f0_cpg_mssr_cfg {
+	DEVICE_MMIO_ROM; /* Must be first */
+};
+
+struct r8a779f0_cpg_mssr_data {
+	struct rcar_cpg_mssr_data cmn; /* Must be first */
+};
+
+/* NOTE: the array MUST be sorted by module field */
+static struct cpg_clk_info_table core_props[] = {
+	RCAR_CORE_CLK_INFO_ITEM(R8A779F0_CLK_S0D12_PER, RCAR_CPG_NONE,
+				RCAR_CPG_NONE, RCAR_CPG_KHZ(66660)),
+
+	RCAR_CORE_CLK_INFO_ITEM(R8A779F0_CLK_CL16M, RCAR_CPG_NONE,
+				RCAR_CPG_NONE, RCAR_CPG_KHZ(16660)),
+};
+
+/* NOTE: the array MUST be sorted by module field */
+static struct cpg_clk_info_table mod_props[] = {
+	RCAR_MOD_CLK_INFO_ITEM(702, R8A779F0_CLK_S0D12_PER),
+	RCAR_MOD_CLK_INFO_ITEM(704, R8A779F0_CLK_S0D12_PER),
+
+	RCAR_MOD_CLK_INFO_ITEM(915, R8A779F0_CLK_CL16M),
+};
+
+static int r8a779f0_cpg_enable_disable_core(const struct device *dev,
+					   struct cpg_clk_info_table *clk_info,
+					   uint32_t enable)
+{
+	int ret = 0;
+	uint32_t reg;
+
+	enable = !!enable;
+
+	switch (clk_info->module) {
+	default:
+		ret = -ENOTSUP;
+		break;
+	}
```

Comment:
this is odd.

Reply:
same as for last question, for now, not any clocks used by arm32 support has to be enable manually.
however, this will be used as soon as arm64 support will come up.
These methods are alos needed to be filled into `rcar_cpg_mssr_data`
Would it be enough to add a comment in the driver explaining that waiting for this list to be filled ?

Reply:
yeah, otherwise this is going to confuse a lot of reviewers :smile:

Reply:
proposed a fix, commenting these part of code, let me know if it can be enough, waiting for arm64 side PR (adding some clocks to these switch case) to be merged !